pkgs/misc/uboot/default.nix

```diff
@@ -314,6 +314,18 @@ in {
     filesToInstall = ["u-boot.bin"];
   };
 
+  ubootRaspberryPi4_32bit = buildUBoot {
+    defconfig = "rpi_4_32b_defconfig";
+    extraMeta.platforms = ["armv7l-linux"];
+    filesToInstall = ["u-boot.bin"];
+  };
+
+  ubootRaspberryPi4_64bit = buildUBoot {
+    defconfig = "rpi_4_defconfig";
+    extraMeta.platforms = ["aarch64-linux"];
+    filesToInstall = ["u-boot.bin"];
+  };
```

Comment:
I was looking today at similar changes b/c this never got merged to master. I found this config file for RPi 3 & 4 (arm64 == ``aarch64``): https://github.com/u-boot/u-boot/blob/0817daa7606ec55ba551acb0208eaba33a548804/configs/rpi_arm64_defconfig. Would it be worth merging these two configurations to reduce code overhead? I know that'll make this PR take slightly longer b/c more to review, so that might be a second step.

There's also an updated UBoot for Raspberry Pi 3B+: https://github.com/u-boot/u-boot/blob/0817daa7606ec55ba551acb0208eaba33a548804/configs/rpi_3_b_plus_defconfig. Complete speculation, but might fix some of the issues tracked in #66960

Reply:
It should be easy for anyone to use  these configs already with a custom buildUBoot. If it turns out there is a big difference, we can still switch to them or provide additional builds.